packages/fuselage-toastbar/src/ToastBarTimed.tsx

```diff
@@ -11,19 +11,30 @@ const ToastBarTimed = ({
   id,
   message,
 }: ToastBarPayload): ReactElement => {
+  const timeInSeconds = time * 1000;
+  const [shouldPause, setShouldPause] = useState(false);
   const dismissToastMessage = useToastBarDismiss();
 
   useEffect(() => {
-    const timeOut = setTimeout(() => {
-      dismissToastMessage(id);
-    }, time * 1000);
+    const initialTime = Date.now();
+    const interval = setInterval(() => {
+      if (shouldPause) {
+        return;
+      }
```

Comment:
This condition is not going to work since the `setInterval`'s callback closure keeps a copy of the original `shouldPause` value.

https://react.dev/reference/react/useEffect#updating-state-based-on-previous-state-from-an-effect